scim/resource_service_principal.go

```diff
@@ -33,8 +34,12 @@ func (a ServicePrincipalsAPI) Create(rsp User) (sp User, err error) {
 	return sp, err
 }
 
-func (a ServicePrincipalsAPI) Read(servicePrincipalID string) (sp User, err error) {
-	servicePrincipalPath := fmt.Sprintf("/preview/scim/v2/ServicePrincipals/%v", servicePrincipalID)
+func (a ServicePrincipalsAPI) Read(servicePrincipalID string, attributes string) (sp User, err error) {
+	attrs := ""
+	if attributes != "" {
+		attrs = "?attributes=" + attributes
+	}
+	servicePrincipalPath := fmt.Sprintf("/preview/scim/v2/ServicePrincipals/%v%s", servicePrincipalID, attrs)
```

Comment:
@alexott how does this reduce the load? Is there backend logic based on attributes?

Reply:
Yes, some attributes, like, `roles` are requiring multiple additional queries against multiple tables, and it's proportional to the number of roles attached. 

But the biggest issue was in the listing of service principals - it was always trying to pull all attributes for all service principals, and this call was timed out. Now we're doing the listing by fetching only application ID & SP ID and then fetching individual instances one by one.

Reply:
I see, thanks for the clarification.